01-30-shield-blocking/actors/player/Player.gd

```diff
@@ -0,0 +1,78 @@
+extends KinematicBody2D
+class_name Player
+
+onready var invencible_timer : Timer = $InvencibleTimer
+onready var animated_sprite : AnimatedSprite = $AnimatedSprite
+onready var animation_player : AnimationPlayer = $AnimationPlayer
+onready var shield : Node2D = $Shield
+
+export var move_speed : = 150.0
+export var jump_force : = 200.0
+export var gravity : = 25.0
+
+var velocity : = Vector2()
+var blocking : = false
+
+
+func _process(delta: float) -> void:
+	if Input.is_action_just_pressed("defend"):
+		shield.appear(not animated_sprite.flip_h)
+		blocking = true
+		move_speed *= 0.5
```

Comment:
This is something I'd like us to avoid for tutorials. Don't mutate variables in place, especially not floats. Have 2 speed constants and assign them to the move_speed instead